system/jlib/jqueue.tpp

```diff
@@ -594,7 +593,7 @@ public:
         PARENT::ensure(limit);
     }
 
-    void stop(unsigned maxReaders = 0, unsigned maxWriters = 0) // stops all waiting operations
+    void stop(unsigned maxReaders, unsigned maxWriters) // stops all waiting operations
     {
         //Assume maxreaders < limit, maxwriters < limit if not provided
```

Comment:
there might not be any relationship between limit and maxReaders/maxWriters,
should this comment and the defaulting to 'limit' if either are 0 be removed too?

Also, out of interest, is there any cost associated with signalling a semaphore with a high number, when it is about to be destroyed?

Reply:
Only that each signal will be done individually, so it would perm several thousand atomic incs in this case.